pkg/tiller/release_rollback.go

```diff
@@ -78,7 +78,11 @@ func (s *ReleaseServer) prepareRollback(req *services.RollbackReleaseRequest) (*
 
 	previousVersion := req.Version
 	if req.Version == 0 {
-		previousVersion = currentRelease.Version - 1
+		p := s.lastSupersededRelease(req.Name)
```

Comment:
should this handle a case where a rollback is attempted on a release in rev 1? The release status would be `deployed`,  so `lastSupersededRelease` will return -1 and the rollback command will fail. An alternative could be to set `previousVersion` to 1 if `currentRelease.Version` is 1. The bad thing about it is that the rollback behavior would differ for revision 1 (create a new revision, even if rev 1 failed) and revisions > 1 (rollback to the last working revision). Maybe skipping the rollback is the right thing to do if there's no revision to rollback to (whether it's because the release is on rev 1 or all releases failed).